### PR TITLE
feat: add blob-explorer crate

### LINKED ISSUE #31: Add blobscan client
## TODO
add simple client for https://sepolia.blobscan.com/

swagger:

https://api.blobscan.com/

### PR DESCRIPTION
**Pull request by mattsse**
closes #31

support for most useful endpoints:
* blocks/id
* transactions/id

types will most likely be changed a bit because some of them can be potentially unified for multiple endpoints

### CODE REVIEW

File: crates/blob-explorers/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+            .query(&query)
+            .send()
+            .await?
+            .json()
+            .await
+    }
+
+    /// Retrieves the __full__ transaction details for given block transaction hash.
+    ///
+    /// This
```
Comment (onbjerg) [inline crates/blob-explorers/src/lib.rs:148]: this..?

File: crates/blob-explorers/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+            .query(&query)
+            .send()
+            .await?
+            .json()
+            .await
+    }
+
+    /// Retrieves the __full__ block details for given block number or hash.
+    ///
+    /// This
```
Comment (onbjerg) [inline crates/blob-explorers/src/lib.rs:197]: this..?

File: crates/blob-explorers/src/request.rs [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+use serde::{ser::SerializeStruct, Deserialize, Serialize};
+
+/// Additional query parameters for the `blocks` endpoint.
+///
+/// By default all fields are requested.
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
+pub struct GetBlockQuery {
+    /// What kind of block to fetch.
+    // TODO flatten this into fields
```
Comment (onbjerg) [inline crates/blob-explorers/src/request.rs:11]: marking the todo

File: crates/blob-explorers/src/request.rs [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+    pub kind: BlockKind,
+}
+
+impl GetBlockQuery {
+    /// Create a new request with the given expansion options.
+    pub const fn new(expand: BlockExpansion) -> Self {
+        Self { expand, kind: BlockKind::Canonical }
+    }
+
+    /// Request transaction,blob and blob_data in the response.
```
Comment (onbjerg) [inline crates/blob-explorers/src/request.rs:23]:
```suggestion
    /// Request transaction, blob and blob data in the response.
```
nit

File: crates/blob-explorers/src/request.rs [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+    pub const fn with_all(self) -> Self {
+        Self { expand: BlockExpansion::all(), ..self }
+    }
+
+    /// Exclude transactions,blob and blob_data in the response.
```
Comment (onbjerg) [inline crates/blob-explorers/src/request.rs:28]:
```suggestion
    /// Exclude transactions, blob and blob data in the response.
```
nit

**Review by onbjerg — APPROVED** [review]
smol nits

**Comment by mattsse** [comment]
merging this so we can iterate on it

**Pull request merged**